NGitLab.Tests/CompareTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Threading.Tasks;
+using NGitLab.Models;
+using NGitLab.Tests.Docker;
+using NUnit.Framework;
+
+namespace NGitLab.Tests
+{
+    public class CompareTests
+    {
+        [Test]
+        [NGitLabRetry]
+        public async Task Test_compare_equal()
+        {
+            using var context = await GitLabTestContext.CreateAsync();
+            var project = context.CreateProject(initializeWithCommits: true);
+            var compareResults = context.Client.GetRepository(project.Id).Compare(new CompareQuery(project.DefaultBranch, project.DefaultBranch));
+
+            Assert.IsNotNull(compareResults);
+            Assert.IsTrue(compareResults.Commits.Length == 0);
+        }
+
+        [Test]
+        [NGitLabRetry]
+        public async Task Test_compare()
+        {
+            using var context = await GitLabTestContext.CreateAsync();
+            var project = context.CreateProject(initializeWithCommits: true);
+
+            context.Client.GetRepository(project.Id).Files.Create(new FileUpsert
+            {
+                Branch = "devtest",
```

Comment:
Tests are failing because you're trying to commit to a missing branch. You need to create the branch beforehand.

Reply:
fixed